tc/sgx/trusted_worker_manager/enclave_wrapper/avalon_enclave_bridge/CMakeLists.txt

```diff
@@ -0,0 +1,146 @@
+ELSE()
+    OPTION(SGX_USE_SIMULATOR "Use the sgx simulator" FALSE)
+ENDIF()
+
+SET(SGX_SDK "$ENV{SGX_SDK}")
+if("${SGX_SDK} " STREQUAL " ")
+    message(FATAL_ERROR "SGX_SDK environment variable not defined!")
+endif()
+SET(SGX_SSL "$ENV{SGX_SSL}")
+if("${SGX_SSL} " STREQUAL " ")
+    SET(SGX_SSL "/opt/intel/sgxssl")
+    message(STATUS "Setting default SGX_SSL=${SGX_SSL}")
+endif()
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+# ADDITIONAL LIBRARIES
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+# sgx edger is a utility that generates edl artifacts
+SET(SGX_EDGER "${SGX_SDK}/bin/x64/sgx_edger8r")
+
+SET(SGX_INCLUDE "${SGX_SDK}/include")
+SET(SGX_SSL_INCLUDE "${SGX_SSL}/include")
+
+SET(SGX_LIBS_UNTRUSTED_NAMES sgx_ukey_exchange)
+
+SET(SGX_LIBS_UNTRUSTED_NAMES ${SGX_LIBS_UNTRUSTED_NAMES} pthread)
+
+IF(${SGX_USE_SIMULATOR})
+    ADD_DEFINITIONS(-D SGX_SIMULATOR)
+    SET(SGX_LIBS_UNTRUSTED_NAMES sgx_urts_sim sgx_uae_service_sim ${SGX_LIBS_UNTRUSTED_NAMES})
+ELSE()
+    SET(SGX_LIBS_UNTRUSTED_NAMES sgx_urts sgx_uae_service ${SGX_LIBS_UNTRUSTED_NAMES})
+ENDIF()
+
+FOREACH(lib ${SGX_LIBS_UNTRUSTED_NAMES})
+    SET(${lib} ${lib})
+ENDFOREACH(lib)
+
+FOREACH(lib ${SGX_LIBS_UNTRUSTED_NAMES})
+    SET(SGX_LIBS_UNTRUSTED ${SGX_LIBS_UNTRUSTED} ${${lib}})
+ENDFOREACH(lib)
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+FUNCTION(SGX_EDGE_UNTRUSTED EDL EDGE_FILES)
+    GET_FILENAME_COMPONENT(EDL_BASE_NAME ${EDL} NAME_WE)
+    GET_FILENAME_COMPONENT(EDL_DIR_NAME ${EDL} DIRECTORY)
+    INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
+
+    SET (EDGE_FILES_LIST "${CMAKE_CURRENT_BINARY_DIR}/${EDL_BASE_NAME}_u.h"
+        "${CMAKE_CURRENT_BINARY_DIR}/${EDL_BASE_NAME}_u.c")
+    SET (${EDGE_FILES} ${EDGE_FILES_LIST} PARENT_SCOPE)
+    ADD_CUSTOM_COMMAND( OUTPUT ${EDGE_FILES_LIST}
+        COMMAND "${SGX_EDGER}" --untrusted ${EDL} --search-path ${SGX_INCLUDE}
+        --search-path ${SGX_SSL_INCLUDE} --search-path ${EDL_DIR_NAME}
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+        DEPENDS ${EDL}
+    )
+ENDFUNCTION()
+
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+FILE(GLOB PROJECT_HEADERS *.h)
+FILE(GLOB PROJECT_SOURCES *.cpp)
+FILE(GLOB PROJECT_EDL ${TCF_CORE_DIR}/trusted_worker_manager/enclave/enclave.edl)
+SOURCE_GROUP("Source" FILES ${PROJECT_HEADERS} ${PROJECT_SOURCES}) 
+
+INCLUDE_DIRECTORIES(${SGX_INCLUDE})
+INCLUDE_DIRECTORIES(${TCF_CORE_DIR}/common)
+INCLUDE_DIRECTORIES(${TCF_CORE_DIR}/common/crypto)
+INCLUDE_DIRECTORIES(${TCF_CORE_DIR}/common/packages/base64)
+
+SGX_EDGE_UNTRUSTED(${PROJECT_EDL} PROJECT_EDGE_SOURCES)
+
+SOURCE_GROUP("Generated" FILES ${PROJECT_EDGE_SOURCES})
+
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m64")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fstack-protector")
```

Comment:
I don't think we need stack-protector for untrusted part. Please check

Reply:
Removed.